build-python.sh

```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
```

Comment:
please revert this. This script is very intentionally using `sh`, so that it's as portable as possible (`bash` on Windows has some eccentricities we'd prefer to avoid).

I'll wait to review the rest of the changes on this file until after you've done that, in case some of them were just a result of switching to bash.